django_lightweight_queue/management/commands/queue_runner.py

```diff
@@ -1,62 +1,81 @@
-                "A value for 'queue-workers' without a value for 'only-queue' "
-                "has no meaning.",
-            )
+class Command(BaseCommand):
+    def add_arguments(self, parser):
+        parser.add_argument('--pidfile', action='store', dest='pidfile', default=None,
+            help="Fork and write pidfile to this file.")
+        parser.add_argument('--logfile', action='store', dest='logfile', default=None,
+            help="Log to the specified file.")
+        parser.add_argument('--touchfile', action='store', dest='touchfile', default=None,
+            help="touch(1) the specified file after running a job.")
+        parser.add_argument('--machine', action='store', dest='machine_number', default=None,
+            help="Machine number, for parallelism")
+        parser.add_argument('--of', action='store', dest='machine_count', default=None,
+            help="Total number of machines running the queues")
+        parser.add_argument('--only-queue', action='store', default=None,
+            help="Only run the given queue, useful for local debugging")
+        parser.add_argument('--config', action='store', default=None,
+            help="The path to an additional django-style config file to load")
+        parser.add_argument('--exact-configuration', action='store_true',
+            help="Run queues on this machine exactly as specified. Requires the"
+                 " use of the '--config' option in addition. It is an error to"
+                 " use this option together with either '--machine' or '--of'.")
+
+    def validate_and_normalise(self, options):
+        if options['exact_configuration']:
+            if not options['config']:
+                raise CommandError(
+                    "Must provide a value for '--config' when using "
+                    "'--exact-configuration'.",
+                )
+
+            invalid_others = ('machine_count', 'machine_number', 'only_queue')
+
+            if any(options[name] is not None for name in invalid_others):
+                raise CommandError(
+                    "Must not specify '--machine', '--of' or '--only-queue'"
+                    " when using '--exact-configuration'.",
+                )
 
-        try:
-            num_only_queue_workers = int(options['num_queue_workers'])
-        except TypeError:
-            num_only_queue_workers = None
         else:
-            if num_only_queue_workers == 0:
-                raise CommandError("Nothing to do! (queue-workers is zero)")
-
-            if num_only_queue_workers < 0:
-                raise CommandError("Cannot have negative queue-workers")
+            real_defaults = {
+                'machine_count': 1,
+                'machine_number': 1,
+            }
+
+            for name, default in real_defaults.items():
+                try:
+                    options[name] = int(options[name])
+                except TypeError:
+                    options[name] = default
```

Comment:
No, this code is changing the 'default' values of the arguments in question from being `None` to being the values from `real_defaults`. This is needed as I've changed them from their previous values in order to detect the difference between:
```
queue_runner --machine 1 --of 1 --exact-configuration --config=bees.py
```
and
```
queue_runner --exact-configuration --config=bees.py
```
As far as I'm aware, with the defaults in argparse being `'1'`, there would not be a way to distinguish the cases. By using `None` as the argparse default (and then replacing that here) we can tell the difference.

Ideally I'd add some tests around this, but I've no idea how to start building tests in a django contrib.